ddtrace/internal/writer.py

```diff
@@ -451,11 +462,11 @@ def flush_queue(self, raise_exc=False):
 
             try:
                 self._retry_upload(self._send_payload, encoded, n_traces)
-            except tenacity.RetryError as e:
+            except Exception as e:
                 self._metrics_dist("http.errors", tags=["type:err"])
                 self._metrics_dist("http.dropped.bytes", len(encoded))
                 self._metrics_dist("http.dropped.traces", n_traces)
-                if raise_exc:
+                if isinstance(e, tenacity.RetryError) and raise_exc:
```

Comment:
I am not sure if we would prefer to remove `isinstance()` and use two separate (but almost identical) exception handlers for tenacity.RetryError and all other Exceptions. I made this change to avoid code duplication but it feels like an anti-pattern

Reply:
well... this is tricky, since before your change any exception other than `tenacity.RetryError` would actually raise and crash the thread.

with your change, no exception will crash the thread. This is probably the thing we want to decide if we want.

regardless of which way we go I think you can remove this `isinstance` check.